stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+
+    Parameters
+    ----------
+    params_list:
+        A list in order of the parameters to be used.
+
+    prior_dict:
+        A dictionary of the priors of parameters to be used.
+
+    kernel_type:
+        The type of kernel to be used in the model.
+
+    mean_type:
+        The type of mean to be used in the model.
+
+    **kwargs:
+        The keyword arguments to be used in the log likelihood function.
+        **Note**: The keyword arguments Times and counts are necessary for
+        calculating the log likelihood.
+        Times: np.array or jnp.array
+            The time array of the lightcurve
+        counts: np.array or jnp.array
+            The photon counts array of the lightcurve
+
+    Returns
+    -------
+    The jaxns specific log likelihood function.
+
+    """
+    if not jax_avail:
+        raise ImportError("Jax is required")
+
+    if not can_make_gp:
+        raise ImportError("Tinygp is required to make the GP model.")
+
+    @jit
+    def likelihood_model(*args):
+        dict = {}
+        for i, params in enumerate(params_list):
+            if params[0:4] == "log_":
+                dict[params[4:]] = jnp.exp(args[i])
+            else:
+                dict[params] = args[i]
+        kernel = get_kernel(kernel_type=kernel_type, kernel_params=dict)
+        mean = get_mean(mean_type=mean_type, mean_params=dict)
+        gp = GaussianProcess(kernel, kwargs["Times"], mean_value=mean(kwargs["Times"]))
+        return gp.log_probability(kwargs["counts"])
+
+    return likelihood_model
+
+
+class GPResult:
+    """
+    Makes a GPResult object which takes in a Stingray.Lightcurve and samples parameters of a model
+    (Gaussian Process) based on the given prior and log_likelihood function.
+
+    Parameters
+    ----------
+    lc: Stingray.Lightcurve object
+        The lightcurve on which the bayesian inference is to be done
+
+    Other Parameters
+    ----------------
+    time : class: np.array
+        The array containing the times of the lightcurve
+
+    counts : class: np.array
+        The array containing the photon counts of the lightcurve
+
+    """
+
+    def __init__(self, Lc: Lightcurve) -> None:
+        self.lc = Lc
+        self.time = Lc.time
+        self.counts = Lc.counts
+        self.Result = None
+
+    def sample(self, prior_model=None, likelihood_model=None, max_samples=1e4):
+        """
+        Makes a Jaxns nested sampler over the Gaussian Process, given the
+        prior and likelihood model
+
+        Parameters
+        ----------
+        prior_model: jaxns.prior.PriorModelType object
+            A prior generator object
+
+        likelihood_model: jaxns.types.LikelihoodType object
+            A likelihood fucntion which takes in the arguments of the prior
+            model and returns the loglikelihood of the model
+
+        max_samples: int, default 1e4
+            The maximum number of samples to be taken by the nested sampler
+
```

Comment:
Does `JaxNS` have more parameters? If so, could they passed through using `**jax_kwargs` or something like that?

Reply:
`JaxNS` has a few other parameters, but they go into different functions. For now the sampler uses default options, but giving user flexibility to change them is something we would want to add.